pkg/tokenvalidation/bootstrapauthenticator.go

```diff
@@ -0,0 +1,102 @@
+package oauth
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kauthenticator "k8s.io/apiserver/pkg/authentication/authenticator"
+	kuser "k8s.io/apiserver/pkg/authentication/user"
+
+	authorizationv1 "github.com/openshift/api/authorization/v1"
+	userv1 "github.com/openshift/api/user/v1"
+	oauthclient "github.com/openshift/client-go/oauth/clientset/versioned/typed/oauth/v1"
+	bootstrap "github.com/openshift/library-go/pkg/authentication/bootstrapauthenticator"
+)
+
+const ClusterAdminGroup = "system:cluster-admins"
+
+type bootstrapAuthenticator struct {
+	tokens            oauthclient.OAuthAccessTokenInterface
+	getter            bootstrap.BootstrapUserDataGetter
+	validator         OAuthTokenValidator
+	implicitAudiences kauthenticator.Audiences
+}
+
+func NewBootstrapAuthenticator(tokens oauthclient.OAuthAccessTokenInterface, getter bootstrap.BootstrapUserDataGetter, implicitAudiences kauthenticator.Audiences, validators ...OAuthTokenValidator) kauthenticator.Token {
+	return &bootstrapAuthenticator{
+		tokens:            tokens,
+		getter:            getter,
+		validator:         OAuthTokenValidators(validators),
+		implicitAudiences: implicitAudiences,
+	}
+}
+
+func (a *bootstrapAuthenticator) AuthenticateToken(ctx context.Context, name string) (*kauthenticator.Response, bool, error) {
+	token, err := a.tokens.Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
why not the given ctx?

Reply:
Somebody probably overlooked this while wiring contexts in the 1.18 kube rebase. It gets removed in a later commit anyway.